beacon_node/network/src/beacon_processor/worker/sync_methods.rs

```diff
@@ -370,19 +370,68 @@ impl<T: BeaconChainTypes> Worker<T> {
     /// Helper function to process backfill block batches which only consumes the chain and blocks to process.
     fn process_backfill_blocks(
         &self,
-        blocks: Vec<Arc<SignedBeaconBlock<T::EthSpec>>>,
+        downloaded_blocks: Vec<BlockWrapper<T::EthSpec>>,
     ) -> (usize, Result<(), ChainSegmentFailed>) {
-        let blinded_blocks = blocks
-            .iter()
-            .map(|full_block| full_block.clone_as_blinded())
-            .map(Arc::new)
-            .collect();
-        match self.chain.import_historical_block_batch(blinded_blocks) {
+        let total_blocks = downloaded_blocks.len();
+        let available_blocks = match downloaded_blocks
+            .into_iter()
+            .map(|block| {
+                self.chain
+                    .data_availability_checker
+                    .check_availability(block)
+            })
+            .collect::<Result<Vec<_>, _>>()
+        {
+            Ok(blocks) => blocks
+                .into_iter()
+                .filter_map(|maybe_available| match maybe_available {
+                    MaybeAvailableBlock::Available(block) => Some(block),
+                    MaybeAvailableBlock::AvailabilityPending(_) => None,
+                })
+                .collect::<Vec<_>>(),
+            Err(e) => match e {
+                AvailabilityCheckError::KzgVerificationFailed => {
+                    return (
+                        0,
+                        Err(ChainSegmentFailed {
+                            // TODO: check this is the right penalty
+                            peer_action: Some(PeerAction::Fatal),
```

Comment:
Seems like we tend to use `LowTolerenceError` for similar sorts of failures

Reply:
Not sure about this one though. Wouldn't this imply they sent us invalid blobs? This seems like something you would do in an attack as it doesn't take a lot to use up a lot of bandwidth. Not sure though..

Reply:
I agree with Mark that this should be Fatal and an outright ban as the peer is clearly malicious in sending us invalidated messages.
From looking at other uses of `PeerAction::Fatal`, it seems to be used for a peer not confirming to the lower level protocol (e.g. Gossipsub/ping/status not supported) or the peer not being useful to us at that moment (e.g. peer does not have resource available while we are syncing)

But i'm not completely sure why we don't use `Fatal` for other clearly invalid stuff 🤔 e.g. 
https://github.com/sigp/lighthouse/blob/c547a11b0da48db6fdd03bca2c6ce2448bbcc3a9/beacon_node/network/src/beacon_processor/worker/gossip_methods.rs#L828-L843

Reply:
cc @paulhauner do you remember why we used `LowTolerance` instead of `Fatal` in the above example?

Reply:
I guess that we don't use `Fatal` just to give a bit of tolerance. I'd think a significant attack is going to involve sending multiple invalid messages, so we just need to boot them before we're overwhelmed. The risk of being too heavy-handed with peer scoring is network partitioning.

Reply:
Okay we'll mark it as `LowToleranceError` then